File: coding/Leetcode/Biweekly_Contest_132/task_100324.cpp
// https://leetcode.com/contest/biweekly-contest-132/problems/clear-digits/
// Created by omega515 on 08.06.24.
//

#include <bits/stdc++.h>

using namespace std;

class Solution{
public:
    static string clearDigits(string s)
    {
        string answer;
        stack<char> st1, st2;
        int n = s.size();
        for (int i = n - 1; i >= 0; --i)
        {
            st1.push(s[i]);
        }
        char top;
        while (!st1.empty())
        {
            top = st1.top();
            st1.pop();
        }


        return answer;
    }
};

int main()
{
    string s;
    string answer;
    Solution solve;

    s = "abc";
    answer = "abc";
    assert(answer == solve.clearDigits(s));

    s = "cb34";
    answer = "";
    assert(answer == solve.clearDigits(s));

    return 0;
}

File: coding/Leetcode/Biweekly_Leetcode/Biweekly_Contest_132/task_100324.py
class Solution:
    def clearDigits(self, s: str) -> str:
        answer = ""
        st1 = []
        for ch in s[::-1]:
            st1.append(ch)

        st2 = []
        for ch in st1[::-1]:
            top = st1.pop()
            if ord('a') <= ord(top) <= ord('z'):
                st2.append(ch)
            else:
                st2.pop()

        answer = "".join(st2)
        return answer


def main():
    solve = Solution()

    s = "abc"
    answer = "abc"
    assert answer == solve.clearDigits(s)

    s = "cb34"
    answer = ""
    assert answer == solve.clearDigits(s)


if __name__ == "__main__":
    main()
